### PR TITLE
Reuse TransactionBuilder in Fullnode

### PR DESCRIPTION
**Pull request by longbowlu**
This PR adds transaction builder api to fullnodes, including:
1. spin off `TransactionBuilder` in its own crate `sui-transaction-builder` to avoid circular dependencies
2. add `FullNodeTransactionBuilderApi` in fullnode that internally uses `TransactionBuilder`
3. add/update some cluster test cases to use fullnode for transaction construction.

### CODE REVIEW

File: crates/sui-cluster-test/src/lib.rs [hunk 1]
```diff
@@ -85,6 +91,15 @@ impl TestContext {
...
 
+    fn get_fullnode_rpc_url(&self) -> &str {
+        self.cluster.fullnode_url()
+    }
+
+    #[allow(dead_code)]
+    fn get_embedded_transaction_builder(&self) -> &TransactionBuilder {
+        self.get_fullnode_client().transaction_builder()
+    }
+
```
Comment (patrickkuo) [inline crates/sui-cluster-test/src/lib.rs:102]: is this needed?

File: crates/sui-open-rpc/src/generate_json_rpc_spec.rs [hunk 2]
```diff
@@ -82,6 +83,7 @@ async fn main() {
     open_rpc.add_module(EventReadApiOpenRpc::module_doc());
     open_rpc.add_module(GatewayWalletSyncApiImpl::rpc_doc_module());
```
Comment (patrickkuo) [inline crates/sui-open-rpc/src/generate_json_rpc_spec.rs:84]: Lets remove all the gateway endpoints from the RPC spec, TransactionBuilderImpl and FullNodeTransactionBuilderApi is the same API.... including both in the spec will cause confusion

**Review by patrickkuo — COMMENTED** [review]
Overall looks good! Just 2 comments.

File: crates/sui-cluster-test/src/lib.rs (hunk 1 above)
Comment (longbowlu) [reply crates/sui-cluster-test/src/lib.rs:102]: removed

**Review by patrickkuo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**